ui/src/services/UserService/UserService.ts

```diff
@@ -13,54 +13,48 @@ const userDetailsSchema = Yup.object().required().shape({
   login: Yup.string().required(),
 });
 
-const emptySchema = Yup.object().required().shape({});
-
-const registerUser = (params: { login: string; email: string; password: string }) =>
-  axios.post(`${context}/register`, params).then(({ data }) => apiKeySchema.validate(data));
-
-const login = (params: { loginOrEmail: string; password: string }) =>
-  axios.post(`${context}/login`, { ...params, apiKeyValidHours: 1 }).then(({ data }) => apiKeySchema.validate(data));
-
-const logout = (apiKey: string | null) =>
-  _securedRequest(apiKey, {
-    method: "POST",
-    url: `${context}/logout`,
-    data: { apiKey },
-  }).then(({ data }) => emptySchema.validate(data));
+export type UserDetails = Yup.InferType<typeof userDetailsSchema>;
 
-const getCurrentUser = (apiKey: string | null) =>
-  _securedRequest(apiKey, {
-    method: "GET",
-    url: context,
-  }).then(({ data }) => userDetailsSchema.validate(data));
-
-const changeProfileDetails = (apiKey: string | null, params: { email: string; login: string }) =>
-  _securedRequest(apiKey, {
-    method: "POST",
-    url: context,
-    data: params,
-  }).then(({ data }) => emptySchema.validate(data));
-
-const changePassword = (apiKey: string | null, params: { currentPassword: string; newPassword: string }) =>
-  _securedRequest(apiKey, {
-    method: "POST",
-    url: `${context}/changepassword`,
-    data: params,
-  }).then(({ data }) => apiKeySchema.validate(data));
+const emptySchema = Yup.object().required().shape({});
 
-const _securedRequest = (apiKey: string | null, config: AxiosRequestConfig) =>
-  axios.request({
-    headers: {
-      Authorization: `Bearer ${apiKey}`,
-    },
-    ...config,
-  });
+const secureRequest = (apiKey: string): AxiosRequestConfig => ({
+  headers: {
+    Authorization: `Bearer ${apiKey}`,
+  },
+});
 
-export const userService = {
-  registerUser,
-  login,
-  logout,
-  getCurrentUser,
-  changeProfileDetails,
-  changePassword,
-};
+export const login = (payload: Components.Schemas.LoginIN) =>
+  api
+    .getClient<Client>()
+    .then((client) => client.postUserLogin(null, { ...payload, apiKeyValidHours: 1 }))
+    .then(({ data }) => apiKeySchema.validate(data));
+
+export const register = (payload: Components.Schemas.RegisterIN) =>
+  api
+    .getClient<Client>()
+    .then((client) => client.postUserRegister(null, payload))
+    .then(({ data }) => apiKeySchema.validate(data));
+
+export const logout = (apiKey: string) =>
+  api
+    .getClient<Client>()
+    .then((client) => client.postUserLogout(null, { apiKey }, secureRequest(apiKey)))
+    .then(({ data }) => emptySchema.validate(data).then(() => undefined));
+
+export const getCurrentUser = (apiKey: string) =>
```

Comment:
@katekozlowska it still seems that this code is "mechanical", that is, could be auto-generated. Same for the schemas. Is this something that we can do?